python/hsfs/core/feature_monitoring_config.py

```diff
@@ -59,20 +62,24 @@ def __init__(
         feature_view_name: Optional[str] = None,
         feature_view_version: Optional[int] = None,
         href: Optional[str] = None,
-    ):
+        transformation_function_dataset_version: Optional[int] = None,
```

Comment:
I think it's better to stick to `training_dataset_version` instead of renaming it. since we don't have such concept of transformation function dataset.

Alternatively, maybe `transformation_functions_td_dataset_version`, although it's a bit verbose. I think `training_dataset_version`  is fine.